spring-test/src/main/java/org/springframework/test/web/client/match/MockRestRequestMatchers.java

```diff
@@ -79,11 +79,13 @@ public static RequestMatcher requestTo(final Matcher<String> matcher) {
 	/**
 	 * Assert the request URI string.
 	 * @param expectedUri the expected URI
+	 * @param uriVars zero or more URI variables to populate the expected URI
 	 * @return the request matcher
 	 */
-	public static RequestMatcher requestTo(final String expectedUri) {
+	public static RequestMatcher requestTo(final String expectedUri, final Object... uriVars) {
 		Assert.notNull(expectedUri, "'uri' must not be null");
-		return request -> assertEquals("Request URI", expectedUri, request.getURI().toString());
+		String uri = UriComponentsBuilder.fromUriString(expectedUri).buildAndExpand(uriVars).encode().toString();
```

Comment:
This will cause expectedUri to always be encoded which wasn't the case until now. 

One option would be to do this only if uriVars are actually provided but then the method behaves inconsistently with regards to encoding based on whether uriVars are provided or not.

Another option would be to have separate methods -- e.g. `requestTo` for exact string match, `requestToUriTemplate` to expand + encode, and possibly also `requestToUri(String expectedUri, Function<UriBuilder, URI> uriFunction)` for full control (as we've done on the new `WebClient`).